src/main/java/ch/njol/skript/expressions/ExprRandomNumber.java

```diff
@@ -35,66 +35,70 @@
 import ch.njol.util.Kleenean;
 import ch.njol.util.Math2;
 
-/**
- * @author Peter Güttinger
- */
-@Name("Random Number")
-@Description({"A random number or integer between two given numbers. Use 'number' if you want any number with decimal parts, or use use 'integer' if you only want whole numbers.",
+@Name("Random Numbers")
+@Description({"A given amount of random numbers or integers between two given numbers. Use 'number' if you want any number with decimal parts, or use use 'integer' if you only want whole numbers.",
 		"Please note that the order of the numbers doesn't matter, i.e. <code>random number between 2 and 1</code> will work as well as <code>random number between 1 and 2</code>."})
 @Examples({"set the player's health to a random number between 5 and 10",
-		"send \"You rolled a %random integer from 1 to 6%!\" to the player"})
-@Since("1.4")
+		"send \"You rolled a %random integer from 1 to 6%!\" to the player",
+		"set {_chances::*} to 5 random integers between 5 and 96"})
+@Since("1.4, INSERT VERSION (Multiple random numbers)")
 public class ExprRandomNumber extends SimpleExpression<Number> {
+
 	static {
 		Skript.registerExpression(ExprRandomNumber.class, Number.class, ExpressionType.COMBINED,
-				"[a] random (1¦integer|2¦number) (from|between) %number% (to|and) %number%");
+				"[a|%-number%] random (:integer|number)[s] (from|between) %number% (to|and) %number%");
 	}
-	
-	@SuppressWarnings("null")
-	private Expression<? extends Number> lower, upper;
-	
+
+	private Expression<Number> amount, lower, upper;
 	private final Random rand = new Random();
-	
 	private boolean integer;
-	
-	@SuppressWarnings({"unchecked", "null"})
+
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parser) {
-		lower = (Expression<Number>) exprs[0];
-		upper = (Expression<Number>) exprs[1];
-		integer = parser.mark == 1;
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		amount = (Expression<Number>) exprs[0];
+		lower = (Expression<Number>) exprs[1];
+		upper = (Expression<Number>) exprs[2];
+		integer = parseResult.hasTag("integer");
 		return true;
 	}
-	
+
 	@Override
 	@Nullable
-	protected Number[] get(final Event e) {
-		final Number l = lower.getSingle(e);
-		final Number u = upper.getSingle(e);
-		if (u == null || l == null)
-			return null;
-		final double ll = Math.min(l.doubleValue(), u.doubleValue());
-		final double uu = Math.max(l.doubleValue(), u.doubleValue());
+	protected Number[] get(Event event) {
+		Number lowerNumber = lower.getSingle(event);
+		Number upperNumber = upper.getSingle(event);
+		if (upperNumber == null || lowerNumber == null)
+			return new Number[0];
```

Comment:
Are you sure this is the behaviour you want? Somebody doing `set {_chances::*} to 5 random integers between {nothing} and 100` will get no numbers returned.

Reply:
IMO it's better to return nothing here as it will indicate to the users that there is a user error in their code